### PR TITLE
Don't Overflow when printing Thawte Strong Extranet Version

### PR DESCRIPTION
**Pull request by mattcaswell**
When printing human readable info on the Thawte Strong Extranet extension
the version number could overflow if the version number == LONG_MAX. This
is undefined behaviour.

Issue found by OSSFuzz.

### CODE REVIEW

**Comment by richsalz** [comment]
See also #10668 removing support for that.

**Comment by mattcaswell** [comment]
I'm not sure how reliable it is to search a database of public certs for an extension that was intended for use in extranets.

File: crypto/x509/v3_sxnet.c [hunk 1]
```diff
@@ -57,12 +57,24 @@ IMPLEMENT_ASN1_FUNCTIONS(SXNET)
...
     SXNETID *id;
     int i;
-    v = ASN1_INTEGER_get(sx->version);
-    BIO_printf(out, "%*sVersion: %ld (0x%lX)", indent, "", v + 1, v);
+
+    /*
+     * Since we add 1 to the version number to display it, we don't support
+     * LONG_MAX since that would cause on overflow.
+     */
+    if (ASN1_INTEGER_get_int64(&v, sx->version) == 0
```
Comment (kaduk) [inline crypto/x509/v3_sxnet.c:69]: Why the explicit check against zero for the return value?  The man page says it only returns 0 or 1.

**Review by kaduk — APPROVED** [review]

File: crypto/x509/v3_sxnet.c (hunk 1 above)
Comment (mattcaswell) [reply crypto/x509/v3_sxnet.c]: I changed this to a boolean check

**Comment by mattcaswell** [comment]
@kaduk - fixup pushed. Please can you reconfirm?

**Review by kaduk — APPROVED** [review]
Reconfirm +1 -- thanks

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**